File: app/src/main/java/com/yasin/rxjavalearn/ScrollingActivity.java
package com.yasin.rxjavalearn;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.yasin.rxjavalearn.BackgroundHeavyWorkWithRx.BackgroundHeavyWorkActivity;
import com.yasin.rxjavalearn.networkCallWithRx.NetworkingActivity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import io.reactivex.Observable;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;

public class ScrollingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scrolling);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(view -> startActivity(new Intent(this, BackgroundHeavyWorkActivity.class)));
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_scrolling, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            startActivity(new Intent(this, NetworkingActivity.class));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void basic(){

        Disposable disposable = Observable.just("First Item" ,"Second Item")
                .subscribe();
        disposable.dispose();

        Observable.just("First Item" ,"Second Item")
                .subscribeOn(Schedulers.io())
                .doOnNext(new Consumer<String>() {
                    @Override
                    public void accept(String s) throws Exception {
                        Log.e("APP",	"on-next:"	+ Thread.currentThread().getName()	+	":"	+	s);
                    }
                })
                .subscribe();
    }


}

File: app/src/main/java/com/yasin/rxjavalearn/ScrollingActivity.kt
package com.yasin.rxjavalearn

import android.content.Intent
import android.os.Bundle

import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.yasin.rxjavalearn.backgroundHeavyWorkWithRx.BackgroundHeavyWorkActivity
import com.yasin.rxjavalearn.nestedFlatMap.NestedFlatMapActivity
import com.yasin.rxjavalearn.networkCallWithRx.NetworkingActivity

import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers

import android.util.Log
import android.view.Menu
import android.view.MenuItem
import kotlinx.android.synthetic.main.content_scrolling.*

class ScrollingActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scrolling)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val fab = findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener {  }
        background_work.setOnClickListener { startActivity(Intent(this, BackgroundHeavyWorkActivity::class.java))}
        network.setOnClickListener { startActivity(Intent(this, NetworkingActivity::class.java))}
        nested_network.setOnClickListener { startActivity(Intent(this, NestedFlatMapActivity::class.java))}
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_scrolling, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId


        if (id == R.id.action_settings) {
            startActivity(Intent(this, NestedFlatMapActivity::class.java))
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    private fun basic() {

        val disposable = Observable.just("First Item", "Second Item")
                .subscribe()
        disposable.dispose()

        Observable.just("First Item", "Second Item")
                .subscribeOn(Schedulers.io())
                .doOnNext { s -> Log.e("APP", "on-next:" + Thread.currentThread().name + ":" + s) }
                .subscribe()
    }


}
